timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/BatchPaxosAcceptorResource.java

```diff
@@ -83,8 +101,12 @@ public BatchPaxosAcceptorResource(PaxosComponents paxosComponents, AcceptorCache
     }
 
     @Override
+    @POST
+    @Path("latest-sequences-prepared-or-accepted")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
     public AcceptorCacheDigest latestSequencesPreparedOrAccepted(
-            Optional<AcceptorCacheKey> maybeCacheKey,
+            @QueryParam(HttpHeaders.IF_MATCH) Optional<AcceptorCacheKey> maybeCacheKey,
```

Comment:
We probably want this to be nullable `AcceptorCacheKey` - if I'm not wrong under this case, at least with standard Jersey if the cache key is `foo` I think it will be parsed as present `Optional[foo]`, and if it isn't provided it actually gets parsed as present `Optional.empty`

Reply:
The issue isn't jersey, it's feign (this is the contract stuff I was talking about last time). As a result, the `BatchPaxosAcceptorRpcClient` takes in a `@Nullable cacheKey` for the corresponding endpoint.

Reply:
In this current configuration on the fully integrated branch, it works just fine.

Reply:
Right, yeah the behaviour I described above was if the RPC client provided an Optional. 

I'm still not entirely comfortable with this though given that the cached method uses a possibly-null one; it seems that they should be consistent, whichever way we decide